### PR TITLE
Move flaky Batch and Profile tests to on-demand build

### PR DESCRIPTION
**Pull request by cormacpayne**
## Description

Move group of Batch and Profile tests that periodically fail on CI into the on-demand build so they don't affect the regular CI. Still need to determine the root cause of these failures.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#making-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by maddieclayton** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/513/

File: src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/BatchApplicationPackageTests.cs [hunk 1]
```diff
@@ -26,7 +26,6 @@ public class BatchApplicationPackageTests : WindowsAzure.Commands.Test.Utilities
         private const string version = "foo";
 
         [Fact]
-        [Trait(Category.AcceptanceType, Category.CheckIn)]
```
Comment (markcowl) [inline src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/BatchApplicationPackageTests.cs]: Please add a trait ```[Category,AcceptanceType, Category.Flaky]```

File: src/ResourceManager/Profile/Commands.Profile.Test/LongRunningCmdletTests.cs [hunk 2]
```diff
@@ -48,7 +48,6 @@ public LongRunningCmdletTests(ITestOutputHelper output)
         }
 
         [Fact]
-        [Trait(Category.AcceptanceType, Category.CheckIn)]
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile.Test/LongRunningCmdletTests.cs]: are all of these flaky?  I think only 2 or 3 of these actually fail (the *Exception* ones), but if not, this is fine.

Comment (cormacpayne) [reply src/ResourceManager/Profile/Commands.Profile.Test/LongRunningCmdletTests.cs]: @markcowl I've found jobs where all of the below tests, with the exception of `CanHandleCmdletStop` and `JobCopiesCmdletParameterSet`, have failed at least one recent build, so I disabled all tests in the class for investigation.

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/538/

**Pull request merged**